File: scripts/migrate-colors.js
const fs = require('fs');
const path = require('path');

function walkDir(dir, callback) {
  if (!fs.existsSync(dir)) return;
  fs.readdirSync(dir).forEach(f => {
    const dirPath = path.join(dir, f);
    const isDirectory = fs.statSync(dirPath).isDirectory();
    isDirectory ? walkDir(dirPath, callback) : callback(dirPath);
  });
}

const dir = path.join(__dirname, 'src');

walkDir(dir, (filePath) => {
  if (filePath.endsWith('.tsx') || filePath.endsWith('.ts')) {
    let content = fs.readFileSync(filePath, 'utf-8');
    let original = content;

    // 1. Replace Hex Codes
    content = content.replace(/#00d4ff/gi, '#735fe9');
    content = content.replace(/#0891b2/gi, '#5a3ee1');

    // 2. Replace 'cyan' with 'brand'
    // Matches cyan-xx, or just the word cyan in strings
    content = content.replace(/\bcyan\b/gi, 'brand');

    if (content !== original) {
      fs.writeFileSync(filePath, content, 'utf-8');
      console.log(`Replaced colors in: ${filePath}`);
    }
  }
});

File: scripts/migrate_colors.py
import os
import re

def replace_in_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    original = content
    # Hex codes
    content = re.sub(r'(?i)#00d4ff', '#735FE9', content)
    content = re.sub(r'(?i)#0891b2', '#5a3ee1', content)
    
    # "cyan" word
    content = re.sub(r'(?i)\bcyan\b', 'brand', content)

    if content != original:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content)
        print(f"Updated {filepath}")

for root, dirs, files in os.walk('src'):
    for file in files:
        if file.endswith('.tsx') or file.endswith('.ts'):
            replace_in_file(os.path.join(root, file))

# Also run on global app files just in case
if os.path.exists('src/app/globals.css'):
    replace_in_file('src/app/globals.css')
